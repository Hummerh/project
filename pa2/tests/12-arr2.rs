fn foo() -> [[i32;1]; 4] { [[1], [2], [3], [4]] }

fn main() {
      let x : i32 = foo()[2][0];
}
