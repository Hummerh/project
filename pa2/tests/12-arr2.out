(crate
    (items
        (fn-def
            (id
                (foo))
            (type-arr
                (type-arr
                    (type-i32)
                    (lit-dec))
                (lit-dec))
            (block
                (arr
                    (exprs
                        (arr
                            (exprs
                                (lit-dec)))
                        (arr
                            (exprs
                                (lit-dec)))
                        (arr
                            (exprs
                                (lit-dec)))
                        (arr
                            (exprs
                                (lit-dec)))))))
        (fn-def
            (id
                (main))
            (block
                (let
                    (pat-id
                        (id
                            (x)))
                    (type-i32)
                    (arr-index
                        (arr-index
                            (fn-call
                                (id
                                    (foo))
                                ())
                            (lit-dec))
                        (lit-dec)))))))